pages/api/browser/auth.ts

```diff
@@ -2,11 +2,12 @@ import cookie from "cookie";
 import type { NextApiRequest, NextApiResponse } from "next";
 import { pingTelemetry } from "ui/utils/replay-telemetry";
 
-const getQueryValue = (query: string | string[]) => (Array.isArray(query) ? query[0] : query);
-const getAppUrl = (path: string) =>
+const getQueryValue = (query: string | string[]): string =>
+  Array.isArray(query) ? query[0] : query;
+const getAppUrl = (path: string): string =>
   `${process.env.NEXT_PUBLIC_APP_URL || process.env.NEXT_PUBLIC_VERCEL_URL}${path}`;
 
-async function initAuthRequest(key: string) {
+async function initAuthRequest(key: string): Promise<unknown> {
```

Comment:
Ends up being `any` because of internal implementation with untyped `response.json()`.

It's a draft PR either way, will see how to handle probably dozens of files that need additional return types in a smart way that isn't purely manual.